File: src/main/java/gameoflife/algorithm/Boundary.java
package gameoflife.algorithm;

public class Boundary extends Cell {
    public Boundary(int x, int y) {
        super(x, y);
    }

    boolean isInBound(Cell that) {
        return checkBoundary(that.x, x) && checkBoundary(that.y, y);
    }

    private static boolean checkBoundary(int value, int boundary) {
        return 0 <= value && value < boundary;
    }

}

File: src/main/java/gameoflife/app/Boundary.kt
package gameoflife.app

import gameoflife.algorithm.Cell

class Boundary(x: Int, y: Int) : Cell(x, y) {

    fun isInBound(that: Cell) = checkBoundary(that.x, x) && checkBoundary(that.y, y)

    private fun checkBoundary(value: Int, boundary: Int) = value in 0 until boundary
}

